components/alibi-explain-server/alibiexplainer/explainer.py

```diff
@@ -0,0 +1,165 @@
+from alibiexplainer.proto import prediction_pb2
+from alibiexplainer.proto import prediction_pb2_grpc
+from alibi.api.interfaces import Explanation
+import grpc
+import tensorflow as tf
+import alibiexplainer.seldon_http as seldon
+import requests
+import os
+from alibiexplainer.model import ExplainerModel
+from tensorflow import keras
+
+SELDON_LOGLEVEL = os.environ.get('SELDON_LOGLEVEL', 'INFO').upper()
+logging.basicConfig(level=SELDON_LOGLEVEL)
+GRPC_MAX_MSG_LEN = 1000000000
+KFSERVING_PREDICTOR_URL_FORMAT = "http://{0}/v1/models/{1}:predict"
+SELDON_PREDICTOR_URL_FORMAT = "http://{0}/api/v0.1/predictions"
+
+class Protocol(Enum):
+    tensorflow_http = "tensorflow.http"
+    seldon_http = "seldon.http"
+    seldon_grpc = "seldon.grpc"
+
+    def __str__(self):
+        return self.value
+
+class ExplainerMethod(Enum):
+    anchor_tabular = "AnchorTabular"
+    anchor_images = "AnchorImages"
+    anchor_text = "AnchorText"
+    kernel_shap = "KernelShap"
+    integrated_gradients = "IntegratedGradients"
+
+    def __str__(self):
+        return self.value
+
+
+class AlibiExplainer(ExplainerModel):
+    def __init__(self,
+                 name: str,
+                 predictor_host: str,
+                 method: ExplainerMethod,
+                 config: Mapping,
+                 explainer: object = None,
+                 protocol: Protocol = Protocol.seldon_grpc,
+                 tf_data_type: str = None,
+                 keras_model: keras.Model = None ):
+        super().__init__(name)
+        self.predictor_host = predictor_host
+        logging.info("Predict URL set to %s", self.predictor_host)
+        self.method = method
+        self.protocol = protocol
+        self.tf_data_type = tf_data_type
+        logging.info("Protocol is %s",str(self.protocol))
+
+        if self.method is ExplainerMethod.anchor_tabular:
+            self.wrapper:ExplainerWrapper = AnchorTabular(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.anchor_images:
+            self.wrapper = AnchorImages(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.anchor_text:
+            self.wrapper = AnchorText(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.kernel_shap:
+            self.wrapper = KernelShap(self._predict_fn, explainer, **config)
+        elif self.method is ExplainerMethod.integrated_gradients:
+            self.wrapper = IntegratedGradients(keras_model, **config)
+        else:
+            raise NotImplementedError
+
+    def load(self):
+        pass
+
+    def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
+        print(arr)
+        if type(arr) == list:
+            arr = np.array(arr)
+        if self.protocol == Protocol.seldon_grpc:
+            return self._grpc(arr)
+        elif self.protocol == Protocol.seldon_http:
+            payload = seldon.create_request(arr, seldon.SeldonPayload.NDARRAY)
```

Comment:
Why not use the SeldonClient, is the current request data structre not supported?

Reply:
As its internal calls easier to use requests directly

Reply:
Ok makes sense, the seldonclient is due a rewrite